### PR TITLE
chore: Supports Atlas Resource Policies

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Supports Atlas Resource Policies

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .changelog/2585.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:new-resource
+resource/mongodbatlas_resource_policy:
```
Comment (lantoli) [inline .changelog/2585.txt]:
```suggestion
resource/mongodbatlas_resource_policy
```

**Review by lantoli — APPROVED** [review]

**Review by jwilliams-mongo — APPROVED** [review]

File: internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go [hunk 2]
```diff
@@ -281,11 +281,9 @@ func authorizeRole(ctx context.Context, client *admin.APIClient, d *schema.Resou
 		roleID = targetRole.GetId()
 	}
 
-	var role *admin.CloudProviderAccessRole
 	var err error
-
 	for i := 0; i < 3; i++ {
-		role, _, err = client.CloudProviderAccessApi.AuthorizeCloudProviderAccessRole(ctx, projectID, roleID, req).Execute()
+		_, _, err = client.CloudProviderAccessApi.AuthorizeCloudProviderAccessRole(ctx, projectID, roleID, req).Execute()
```
Comment (EspenAlbert) [inline internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go]: I don't think this change should be included

**Comment by EspenAlbert** [comment]
Change PR title to `feat`?
Maybe we can do a `merge` commit?

File: internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go]: reverted the change

**Pull request merged**